### PR TITLE
Add CPU tests to the Travis framework.

### PR DESCRIPTION
**Pull request by bartoldeman**
Most still fail, so this PR can be used to track various CPUEMU fixes to do.

(I never thought I'd be programming in Python for DOSEMU but here you go :)

### CODE REVIEW

**Comment by stsp** [comment]
Cool!
Please see if you fixed #948 #949

**Comment by stsp** [comment]
And of course #86

File: test/test_dos.py [hunk 1]
```diff
@@ -4485,6 +4486,74 @@ def test_lfn_file_info_mfs_6GiB(self):
...
+                    self.assertEqual(list(g), lines)
+            except IOError:
+                # copy as reference file
+                with open("test-i386.log", "w") as g:
+                    g.write("".join(lines))
+
+    def test_cpu_1_vm86native(self):
+        """CPU test: native vm86 + native DPMI (i386 only)"""
+        if uname()[4] == 'i686':
+            self._test_cpu("vm86native")
```
Comment (andrewbird) [inline test/test_dos.py]: Perhaps you could revise this to 
```python
if uname()[4] == 'x86_64':
   self.skipTest("x86_64 doesn't support native vm86()")
self._test_cpu("vm86native")
```

Other than that, it looks great, I'm so glad you done this!

**Comment by andrewbird** [comment]
~~~
Test FR-DOS-1.20 CPU test: JIT vm86 + native DPMI                                ... ok
~~~

...

~~~
Test PP-DOS-GIT CPU test: JIT vm86 + native DPMI                                 ... FAIL
~~~

That's interesting, I would have thought that FreeDOS and FDPP results would be identical for all your CPU tests.

**Comment by bartoldeman** [comment]
thanks for the feedback @andrewbird . I'll make that change. I'm surprised about the FDPP failures too, will check myself.

@stsp I'm not sure if the related fixes will fix those bugs but it would be great if they did. If they fail with dosemu1 too then likely not. Remaining sim failures are for looking up page fault eip's and a few FPU failures (it claims that log(2.0)=-inf and log(1.4)=-inf).

**Comment by andrewbird** [comment]
I added KVM support to the travis test environment, it seems to work with your current PR but there are again diffs between FreeDOS output and FDPP. It seems that FDPP just produces two blank lines instead of the 5k+ of FreeDOS, see https://travis-ci.com/andrewbird/dosemu2/builds/151419497 or just the salient part, below

~~~
======================================================================

FAIL: Test PP-DOS-GIT CPU test: JIT vm86 + native DPMI                                

----------------------------------------------------------------------

Traceback (most recent call last):

  File "test/test_dos.py", line 4529, in test_cpu_2_jitnative

    self._test_cpu("emulated", "native", "vm86")

  File "test/test_dos.py", line 4510, in _test_cpu

    self.assertEqual(len(lines), 5056)

AssertionError: 2 != 5056

======================================================================

FAIL: Test PP-DOS-GIT CPU test: KVM vm86 + KVM DPMI                                   

----------------------------------------------------------------------

Traceback (most recent call last):

  File "test/test_dos.py", line 4543, in test_cpu_kvm

    self._test_cpu("kvm", "kvm", "off")

  File "test/test_dos.py", line 4510, in _test_cpu

    self.assertEqual(len(lines), 5056)

AssertionError: 2 != 5056

======================================================================

FAIL: Test PP-DOS-GIT CPU test: KVM vm86 + native DPMI                                

----------------------------------------------------------------------

Traceback (most recent call last):

  File "test/test_dos.py", line 4537, in test_cpu_kvmnative

    self._test_cpu("kvm", "native", "off")

  File "test/test_dos.py", line 4510, in _test_cpu

    self.assertEqual(len(lines), 5056)

AssertionError: 2 != 5056

======================================================================

FAIL: Test PP-DOS-GIT CPU test: simulated vm86 + native DPMI                          

----------------------------------------------------------------------

Traceback (most recent call last):

  File "test/test_dos.py", line 4533, in test_cpu_simnative

    self._test_cpu("emulated", "native", "vm86sim")

  File "test/test_dos.py", line 4510, in _test_cpu

    self.assertEqual(len(lines), 5056)

AssertionError: 2 != 5056

----------------------------------------------------------------------

Ran 328 tests in 891.913s

FAILED (failures=4, skipped=27)

The command "sudo -E su ${USER} -c ./travis_test.sh" exited with 1.

~~~

**Comment by stsp** [comment]
Does fdpp eat some console output?
Any example?

**Comment by bartoldeman** [comment]
The issue with FDPP is that comcom32 does not understand LFNs and the executable is called test-i386.exe. I simply called it test.exe now.

**Comment by bartoldeman** [comment]
Ok, all checks have passed now, so in principle that could be merged and then I can enable the disabled (skipped) fullsim and full tests in the PRs that fix them.

**Comment by stsp** [comment]
Sure, thanks!
This is exactly the kind of work we
need to finaly make a beta out.
Very cool. :)

**Pull request merged**